src/server/editorServices.ts

```diff
@@ -536,15 +530,20 @@ module ts.server {
         updateProjectStructure() {
             this.log("updating project structure from ...", "Info");
             this.printProjects();
+            var openFilesReferenced: ScriptInfo[] = [];
+            var unattachedOpenFiles: ScriptInfo[] = [];
             for (var i = 0, len = this.openFilesReferenced.length; i < len; i++) {
-                var refdFile = this.openFilesReferenced[i];
-                refdFile.defaultProject.updateGraph();
-                var sourceFile = refdFile.defaultProject.getSourceFile(refdFile);
-                if (!sourceFile) {
-                    this.openFilesReferenced = copyListRemovingItem(refdFile, this.openFilesReferenced);
-                    this.addOpenFile(refdFile);
+                var referencedFile = this.openFilesReferenced[i];

```

Comment:
I think a comment above this logic would be helpful; at a glance, I'm a little unsure what the motive is behind separating referenced files and opened files.